api/dbadapters/bigquery.ts

```diff
@@ -190,6 +194,28 @@ export class BigQueryDbAdapter implements IDbAdapter {
 
   public async close() {}
 
+  public async persistedStateMetadata(
+    compiledGraph: dataform.ICompiledGraph
```

Comment:
it definitely feels a bit weird to have an API db adapter depend on the CompiledGraph proto. if we stop depending on core, we can get rid of this dependency